src/qibolab/platforms/platform.py

```diff
@@ -30,14 +35,63 @@ def disconnect(self):
         self.design.disconnect()
         self.is_connected = False
 
-    def execute_pulse_sequence(self, sequence, nshots=1024, relaxation_time=None):
-        if relaxation_time is None:
-            relaxation_time = self.relaxation_time
-        return self.design.play(self.qubits, sequence, nshots=nshots, relaxation_time=relaxation_time)
+    def execute_pulse_sequence(self, sequence, **kwargs):
+        options = ExecutionParameters(kwargs)
+
+        if options.relaxation_time is None:
+            options.relaxation_time = self.relaxation_time
+        if options.fast_reset:
+            options.fast_reset = {
+                qubit.name: self.create_RX_pulse(qubit=qubit.name, start=sequence.finish)
+                for qubit in self.qubits.values()
+                if not qubit.flux_coupler
+            }
+
+        return self.design.play(
+            self.qubits,
+            sequence,
+            options=options,
+        )
+
+    def sweep(self, sequence, *sweepers, **kwargs):
+        options = ExecutionParameters(kwargs)
+
+        if options.relaxation_time is None:
+            options.relaxation_time = self.relaxation_time
+        if options.fast_reset:
+            options.fast_reset = {
+                qubit.name: self.create_RX_pulse(qubit=qubit.name, start=sequence.finish)
+                for qubit in self.qubits.values()
+                if not qubit.flux_coupler
+            }
 
-    def sweep(self, sequence, *sweepers, nshots=1024, relaxation_time=None, average=True):
-        if relaxation_time is None:
-            relaxation_time = self.relaxation_time
         return self.design.sweep(
-            self.qubits, sequence, *sweepers, nshots=nshots, relaxation_time=relaxation_time, average=average
+            self.qubits,
+            sequence,
+            *sweepers,
+            options=options,
         )
+
+
+@dataclass
+class ExecutionParameters:
+    """Data structure to deal with execution parameters
+
+    :nshots: Number of shots per point on the experiment
+    :relaxation_time: Relaxation time for the qubit
+    :fast_reset: Enable or disable fast reset
+    :sim_time: Time for the simulation execution
+    :acquisition_type: Data acquisition mode (SPECTROSCOPY, INTEGRATION, RAW, DISCRIMINATION)
+    :averaging_type: Data averaging mode (CYLIC[True averaging], SINGLESHOT[False averaging], [SEQUENTIAL, bad averaging])
+    """
+
+    nshots: Optional[np.uint32] = None
+    relaxation_time: Optional[np.uint32] = None
+    fast_reset: Optional[bool] = False
+    sim_time: Optional[np.float64] = 10e-6
+    acquisition_type: Optional[str] = None
+    averaging_mode: Optional[str] = None
```

Comment:
For the last two, it may make sense to use custom `Enum` classes instead of strings, see for example the `Parameter` class in sweeper.py. In particular something like
```py
class AcquisitionType(Enum):
    """Types of data acquisition from hardware."""

    RAW = auto()
    INTEGRATION = auto()
    DISCRIMINATION = auto()
```
and you use it like
```py
platform.execute_pulse_sequence(sequence, nshots, acquisition_type=AcquisitionType.DISCRIMINATION)
```
I believe this is more in line with Zurich code too and it makes it easier to document what are the different acquisition types (and averaging modes).

That being said, I am not sure how much these options are relevant for other drivers, but for sure we need a way to acquire raw data on all platforms.

Reply:
It's exactly the same as how it would be written with Zurich.

I think these three options should be quite general. I already took care of the weird SPECTROSCOPY type they have, which is only used on frequency sweeps for the resonator. The averaging modes should also be general where the only difference being that you can only acquire one of them.